uk.co.compucorp.civicrm.hrcore/hrcore.php

```diff
@@ -26,7 +26,10 @@ function hrcore_civicrm_config(&$config) {
  * @param array $tasks
  */
 function hrcore_civicrm_searchTasks($objectName, &$tasks) {
-  if ($objectName !== 'contact' || !CRM_Core_Permission::check('create users')) {
+  $isContact = $objectName === 'contact';
```

Comment:
Sure, I agree, but if we just add functions to this file we'll end up with files like [this](https://github.com/civicrm/civihr/blob/c921837290c61e8c9b102dd3358862969e8c3b78/hrui/hrui.php) with almost 1000 lines. Neither result is optimal. That's why I'd prefer to move the logic to a separate class, so that's what I'll do.